### ISSUE TITLE
Test builds leave Git workspace on master

### ISSUE THREAD

**Issue by AdamBrousseau** [AdoptOpenJDK/openjdk-tests#1895]
**Describe the bug**
A few recent changes have caused our disk usage on master to increase substantially.
1. #1838 **Use ADOPTOPENJDK_REPO and ADOPTOPENJDK_BRANCH param in job configure**. This changed the behaviour of the test job launches to no longer use lightweight checkout. The effect is that we no longer pull the single pipeline script [JenkinsfileBase](https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/buildenv/jenkins/JenkinsfileBase) onto master. But rather we clone the entire test Git repo which totals around 175M (as of today).
2. The compounding factor is the way the parallel tests run. Now that test jobs spawn multiple child jobs, each with a new name, they each leave another copy of the test repo in a workspace on master.

This is where we're at today on one of our internal servers.
PR1838 was merged on June 25. Parallel testing had been running partially for a little while before this if I recall correctly.
June 25th:
2426 Test jobs
Totaling 208GB disk usage on master. This includes both the workspaces and the console log history which we had set at 30.

Today July 15:
1565 Test jobs (we cleaned up alot of stuff)
Totaling 275GB of disk usage **only** for the workspaces on master (no logs included in calculation).

Given that this instance currently is capped at 500GB storage and we have other builds on it, we have very little room right now.

What are our options?
1. Revert back to lightweight checkout. This would break being able to use a parameter variable for the scm checkout (see 1838)
2. Use a custom refspec for the pipeline clones. I'm not sure this would work as we may have multiple use cases which may not all fit into 1 refspec. TBD
3. Use shallow clone. Ie. Don't clone the history of the repo, just the latest of whatever branch we need. This seems like the best option to start testing in my opinion.

**Comment by smlambert** [comment]
Thanks for this investigation and synopsis @AdamBrousseau - I agree that option 3 looks like the best starting point.

**Comment by llxia** [comment]
Thanks @AdamBrousseau . I will update the script to do shallow clone.

**Comment by llxia** [comment]
I am going reopen this issue as git clone shallow does not seem to save much space. I am not sure what exact cmd Jenkins use, but my local git clone only uses 87M vs Jenkins 174M.

Just for the record, I have tried all [cloneOptions in the doc](https://jenkinsci.github.io/job-dsl-plugin/#method/javaposse.jobdsl.dsl.helpers.scm.GitExtensionContext.cloneOptions). No of them seems to help much. Also, lightweight option was removed because it does not work with build parameters (see [open issue](https://issues.jenkins-ci.org/browse/JENKINS-48431))
